apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/account/service/AccountServiceImpl.java

```diff
@@ -117,6 +106,24 @@ public Account getAccount(long id, int height) {
         return account;
     }
 
+    private Account getAccount(DbKey dbKey, int height) {
+        if (height < 0 || blockChainInfoService.doesNotExceed(height)) {
+            return accountTable.get(dbKey);
+        }
+        checkAvailable(height);
+        return accountTable.get(dbKey, height);
+    }
+
+    void checkAvailable(int height) {
```

Comment:
Maybe better to generalize such checks to prevent code duplication and keep blockchain logic consistent?

Reply:
I was thinking about while development. A generic method causes more code with Functions that is not quite clear. Better to leave it as I did here.